src/vm/object.cpp

```diff
@@ -221,6 +221,11 @@ TypeHandle Object::GetGCSafeTypeHandleIfPossible() const
         pMTToCheck = thElem.GetMethodTable();
     }
 
+    if (pMTToCheck == g_pFreeObjectMethodTable)
+    {
+        return NULL;
+    }
+
     Module * pLoaderModule = pMTToCheck->GetLoaderModule();
 
     BaseDomain * pBaseDomain = pLoaderModule->GetDomain();
```

Comment:
You can delete this line while you are on it. `pBaseDomain` is not used anymore.